MAISTRA-2648: WASM fix for s390x

knm3000 opened the pull request:
Fixes https://issues.redhat.com/browse/MAISTRA-2648
The permanent fix is not a part of maistra/envoy yet:
proxy-wasm/proxy-wasm-cpp-host#198

Signed-off-by: Konstantin Maksimov <konstantin.maksimov@ibm.com>

jwendell commented:
/ok-to-test

oschaaf reviewed (approved):
Thanks @knm3000 for moving the change into this repo. 
I have no other comments, LGTM

oschaaf commented:
Note: I tagged this PR with `do-not merge` as I think @twghu should be able to gate this (I also added him as a reviewer) - he may have a large set of changes inbound for the target branch, and it might help if the target doesn't move.

oschaaf commented:
Sorry - I misread the target branch; please disregard the earlier comment. I removed the `do-not-merge` label. Sorry about the noise!

jwendell commented:
/retest

jwendell commented on bazel/repositories.bzl:
```diff
@@ -795,6 +795,10 @@ def _proxy_wasm_cpp_host():
         build_file = "@envoy//bazel/external:proxy_wasm_cpp_host.BUILD",
+        # The patch fixes WASM on s390x https://issues.redhat.com/browse/MAISTRA-2648
+        # The permanent fix is https://github.com/proxy-wasm/proxy-wasm-cpp-host/pull/198
+        patches = ["@envoy//bazel/external:proxy_wasm_cpp_host.patch"],
```
nit: Generally speaking it's a good idea to name the file after its meaning or purpose, or even with the title of the original patch, instead of naming it with the name of the package. What if, in the future there is another patch for this same package?

knm3000 replied:
Renamed the patch file name to `proxy_wasm_s390x.patch`.

knm3000 committed:
MAISTRA-2648: WASM fix for s390x

Fixes https://issues.redhat.com/browse/MAISTRA-2648
The permanent fix is not a part of maistra/envoy yet:
proxy-wasm/proxy-wasm-cpp-host#198

Signed-off-by: Konstantin Maksimov <konstantin.maksimov@ibm.com>

jwendell commented:
/retest

oschaaf commented:
/retest
